server/index.ts

```diff
@@ -106,50 +114,78 @@ router.post('/message', async (req, res) => {
     !msg.space ||
     !msg.payload ||
     Object.keys(msg.payload).length === 0
-  ) return sendError(res, 'wrong signed message');
+  )
+    return sendError(res, 'wrong signed message');
 
   if (!spaces[msg.space] && msg.type !== 'settings')
     return sendError(res, 'unknown space');
 
-  if (!msg.timestamp || typeof msg.timestamp !== 'string' || msg.timestamp > (ts + 300))
+  if (
+    !msg.timestamp ||
+    typeof msg.timestamp !== 'string' ||
+    msg.timestamp > ts + 300
+  )
     return sendError(res, 'wrong timestamp');
 
   if (!msg.version || msg.version !== pkg.version)
     return sendError(res, 'wrong version');
 
-  if (!msg.type || !['proposal', 'vote', 'settings'].includes(msg.type))
+  if (
+    !msg.type ||
+    !['proposal', 'vote', 'settings', 'delete-proposal'].includes(msg.type)
+  )
     return sendError(res, 'wrong message type');
 
-  if (!await verifySignature(body.address, body.sig, hashPersonalMessage(body.msg)))
+  if (
+    !(await verifySignature(
+      body.address,
+      body.sig,
+      hashPersonalMessage(body.msg)
+    ))
+  )
     return sendError(res, 'wrong signature');
 
+  if (msg.type === 'delete-proposal') {
+    let query = 'SELECT address FROM messages WHERE id = ?';
```

Comment:
You should explicitly check for type = 'proposal', otherwise it could be used to delete vote

Reply:
So the query can be changes by the user?

Reply:
Someone could build a signed message to delete a vote of himself. We dont want this, when a proposal ended the results should not change.

Reply:
Like this e0423ba?

Reply:
Yes